### DISCUSSION TITLE
ProductSwitch: Number of Items per Line

### DISCUSSION THREAD

**Discussion by bachlge** [UI5/webcomponents#9733] [Ideas]
[ProductSwitch](https://sap.github.io/ui5-webcomponents/components/fiori/ProductSwitch/), seems to have a hidden Property `desktop-columns`.
The code of the Basic Sample only shows `<ui5-product-switch>` whereas it shows `<ui5-product-switch ui5-product-switch="" desktop-columns="3">` when I inspect the rendered code.
I suggest to make this a public and documented property.
For responsive behaviour, additional or alternative public properties like useAllSpace, minColumns, maxColumns could be condidered.

**Comment by nnaydenow** [comment]
Hi @bachlge,

By specification the columns count is determinated by 2 things:
- items number (if items are > 6 show 4 columns and if they are <= 6 show 3 columns)
- available screen width (desktop based on items count, tablet 3 and mobile 1)

Could you share more motivation why public API should be exposed?

My thoughts: When the component was initially introduced CSS grid wasn't fully supported. Now, I think the component might be reworked and just to expose a part which will allow people to manipulate the grid as they want.

**Reply by bachlge** [reply]
Hi @nnaydenow, 
thx for explaining the spec, i was not aware of this.
In my application, all components are pretty responsive, dynamically using the available space.
Just ProductSwitch is not, that is why I started to dig & found this property which I thought was forgotten in the documentation.
Full responsiveness (considering not only number of items but also available space) would be great :-)

**Reply by bachlge** [reply]
PS. the SAP Fiori Launchpad of our S/4HANA system (Technical Component ID sap.ushell.components.homepage) behaves like I would want ProductSwitch  to behave ...

**Reply by nnaydenow** [reply]
Hi @bachlge,

Sorry for the delayed response. Please open feature request so the respective team for the component to process it.

Thanks!

**Comment by bachlge** [comment]
https://github.com/SAP/ui5-webcomponents/issues/9864
